packages/e2e/cypress/e2e/run/actions-pipelinerun.cy.js

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const namespace = 'e2e-actions';
+describe('Edit and Run Pipeline Run', () => {
+  before(() => {
+    cy.exec('kubectl version --client');
+    cy.exec(`kubectl create namespace ${namespace} || true`);
+  });
+
+  after(() => {
+    cy.exec(`kubectl delete namespace ${namespace} || true`);
+  });
+
+  it('should create pipelinerun', function () {
+    const uniqueNumber = Date.now();
+
+    const pipelineName = `sp-${uniqueNumber}`;
+    const pipeline = `apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: ${pipelineName}
+  namespace: ${namespace}
+spec:
+  tasks:
+    - name: hello
+      taskSpec:
+        steps:
+          - name: echo
+            image: busybox
+            script: |
+              #!/bin/ash
+              echo "Hello World!"
+    `;
+    cy.exec(`echo "${pipeline}" | kubectl apply -f -`);
+    cy.visit(
+      `/#/pipelineruns/create?namespace=${namespace}&pipelineName=${pipelineName}`
+    );
+    cy.get('[id=create-pipelinerun--namespaces-dropdown]').should(
+      'have.value',
+      namespace
+    );
+    cy.get('[id=create-pipelinerun--pipelines-dropdown]').should(
+      'have.value',
+      pipelineName
+    );
+
+    cy.contains('button', 'Create').click();
+    cy.get(`[title*="${pipelineName}-run"]`)
+      .should('have.length', 1, { timeout: 10000 })
+      .parents('tr')
+      .find('[class="tkn--status"]')
+      .each(item => {
+        cy.wrap(item, { timeout: 15000 }).should('contain.text', 'Succeeded');
+      });
+    cy.contains(`${pipelineName}-run`).parent().click();
+
+    cy.contains('button', 'Actions').click();
+    cy.contains('button', 'Edit and Run').click();
+    cy.contains('button', 'Create').click();
```

Comment:
So, I have tried todo that but failed.

`getByRole(/textbox/).textContent;` returns content without space `apiVersion: tekton.dev/v1beta1kind` and cannot be converted to yaml. `getByRole(/textbox/).innerHTML` requires even more manipulations. So only string validation can be used.  In `CreatePipelineRun.test.js` I have done the validation by using regexp.
However, for e2e we have another issue. Tekton has some default parameters https://tekton.dev/docs/pipelines/install/#customizing-basic-execution-parameters, e.g. `timeout: 1h0m0s` are populated by tekton engine one pipeline run creation. Do we want to compare those fields as well? Not sure. One of the options is to get first pipeline run, remove those fields (and fields in our logic), and compare. 
Another approach would be to use after `Create` with `cy.intercept`. In this case we will be able to use json\yaml and compare, but fields removal is required.
Instead of removal, since we don't know all the fields ahead, and they are subject to change, we can construct new pipelinerun with only known fields.
One more possible option is to leverage some advanced testing features \ frameworks for such goal as  skipping unknown fields. However, I am not aware of such, except json schema validation.

And finally the last option would be to use only some contains assertions.That is what I did.

Reply:
> Used only one contains assertions for pipeline name

Yes that's fine, we just want to ensure the content is present and reasonable. Checking the name is enough for now. No need to do a full comparison.